PowerValidatedSolutions.psm1

```diff
@@ -48145,6 +48145,383 @@ Function Test-WMSubnetInput {
+    Try {
+        $respond = Invoke-WebRequest -Uri "https://$caFqdn/certsrv/certrqxt.asp" -Credential $domainCreds -Method Get -SkipCertificateCheck
+    } catch {
+        Debug-ExceptionWriter -object $_
+    }
+
+    if (-Not ($respond.Content -match ".*Microsoft RSA SChannel Cryptographic Provider.*$certificateTemplate.*")) {
+        Write-Error "The Microsoft Certificate Template $certificateTemplate not found." -ErrorAction Stop
+    } else {
+        $CertAttributes = "CertificateTemplate:$certificateTemplate"
+    }
+
+    if ($getCArootCert.IsPresent) {
+        $createRandomCounter = 0
+        for ($createRandomCounter = 0; $createRandomCounter -lt 8; $createRandomCounter++) {
+            $randomString = -join (((48..57)+(65..90)+(97..122)) * 80 |Get-Random -Count 6 |%{[char]$_})
+            $rootCaFilePathTemp = Join-Path -path $certDir -ChildPath "$fileName-$randomString.p7b"
+            if (-Not(Test-Path -Path $rootCaFilePathTemp)) {
+				Write-Output "1" | Set-Content $rootCaFilePathTemp
+				if(Test-Path -Path $rootCaFilePathTemp) {
+					Remove-Item $rootCaFilePathTemp
+					break;
+				}
+			} else {
+				if($createRandomCounter -gt 6) {
+					Write-Error "Error: Unable to create temporary files in directory $certDir." -ErrorAction Stop
+					Exit
+				}
+			}
+		}
+        $rootCaFilename = "$caFqdn-rootCA.pem"
+        $rootCaFilePath = Join-Path -Path $certDir -ChildPath $rootCaFilename
+        if (Test-Path -Path $rootCaFilePath) {
+            Write-Error "Error: $rootCaFilePath file exists.  Please remove or rename the existing file." -ErrorAction Stop
+        }
+        if (-Not ([bool] (Get-Command -ErrorAction Ignore -Type Application openssl))) {
+            Write-Error "Error: Openssl not found.  Please verify openssl is installed."
+        }
+    }
+
+    # Create form array
+    $formFields = @{
+        Mode = "newreq"
+        FriendlyType = "Saved-Request Certificate"
+        CertRequest = Get-Content -Path $csrFilePath -Raw
+        CertAttrib = $CertAttributes
+        TargetStoreFlags = 0
+        SaveCert = "yes"
+    }
+
+    # Request CSR to be signed by MSCA
+    Try {
+        $CertSubmitted = Invoke-WebRequest -Uri "https://$caFqdn/certsrv/certfnsh.asp" -Credential $domainCreds -Method Post -Body $formFields -SkipCertificateCheck
+        $RequestID = ($CertSubmitted.content -split '\n' | Where-Object -FilterScript {$_ -match "certnew.cer\?ReqID=[0-9]"}) -replace '[^0-9]'
+        $null = Invoke-WebRequest -Uri "https://$caFqdn/certsrv/certnew.cer?ReqID=${RequestID}&Enc=b64" -Credential $domainCreds -OutFile $crtFilePath -SkipCertificateCheck
+        if ($getCArootCert.IsPresent) {
+            $null = Invoke-WebRequest -Uri "https://$caFqdn/certsrv/certnew.p7b?ReqID=CACert&Enc=b64" -Credential $domainCreds -OutFile $rootCaFilePathTemp -SkipCertificateCheck
+            openssl pkcs7 -inform PEM -outform PEM -in $rootCaFilePathTemp -print_certs > $rootCaFilePath
+        }
+    } catch {
```

Comment:
```suggestion
    } Catch {
```

Reply:
Addressed in commit 53039acbaaffafaf219274cac1a3684735130f53